src/main/java/org/wildfly/security/auth/provider/ldap/OtpCredentialLoader.java

```diff
@@ -30,16 +30,21 @@
  */
 public class OtpCredentialLoader implements CredentialLoader, CredentialPersister {
 
+    public static final String DEFAULT_CREDENTIAL_NAME = "otp";
+
+    private final String myCredentialName; // name of credential defined by following LDAP attributes
     private final String algorithmAttributeName;
     private final String hashAttributeName;
     private final String seedAttributeName;
     private final String sequenceAttributeName;
 
-    public OtpCredentialLoader(String algorithmAttributeName, String hashAttributeName, String seedAttributeName, String sequenceAttributeName) {
+    public OtpCredentialLoader(String credentialName, String algorithmAttributeName, String hashAttributeName, String seedAttributeName, String sequenceAttributeName) {
+        Assert.assertNotNull(credentialName);
```

Comment:
I know you didn't write this code but assertNotNull should never be used to validate a parameter to a public constructor or method.  Use checkNotNullParam instead.